fastplotlib/graphics/_base.py

```diff
@@ -69,123 +79,266 @@ def __init__(
+    def name(self, value: str):
+        self._name.set_value(self, value)
+
+    @property
+    def offset(self) -> np.ndarray:
+        """Offset position of the graphic, array: [x, y, z]"""
+        return self._offset.value
+
+    @offset.setter
+    def offset(self, value: np.ndarray | list | tuple):
+        self._offset.set_value(self, value)
+
+    @property
+    def rotation(self) -> np.ndarray:
+        """Orientation of the graphic as a quaternion"""
+        return self._rotation.value
+
+    @rotation.setter
+    def rotation(self, value: np.ndarray | list | tuple):
+        self._rotation.set_value(self, value)
+
+    @property
+    def visible(self) -> bool:
+        """Whether the graphic is visible"""
+        return self._visible.value
+
+    @visible.setter
+    def visible(self, value: bool):
+        self._visible.set_value(self, value)
+
+    @property
+    def deleted(self) -> bool:
+        """used to emit an event after the graphic is deleted"""
+        return self._deleted.value
 
-        if not isinstance(name, str):
-            raise TypeError("`Graphic` name must be of type <str>")
+    @deleted.setter
+    def deleted(self, value: bool):
+        self._deleted.set_value(self, value)
 
-        if self._plot_area is not None:
-            self._plot_area._check_graphic_name_exists(name)
+    @property
+    def block_events(self) -> bool:
+        """Used to block events for a graphic and prevent recursion."""
+        return self._block_events
 
-        self._name = name
+    @block_events.setter
+    def block_events(self, value: bool):
+        self._block_events = value
 
     @property
-    def world_object(self) -> WorldObject:
+    def world_object(self) -> pygfx.WorldObject:
         """Associated pygfx WorldObject. Always returns a proxy, real object cannot be accessed directly."""
         # We use weakref to simplify garbage collection
         return weakref.proxy(WORLD_OBJECTS[self._fpl_address])
 
-    def _set_world_object(self, wo: WorldObject):
+    def _set_world_object(self, wo: pygfx.WorldObject):
         WORLD_OBJECTS[self._fpl_address] = wo
 
-    @property
-    def position(self) -> np.ndarray:
-        """position of the graphic, [x, y, z]"""
-        return self.world_object.world.position
+        self.world_object.visible = self.visible
 
-    @property
-    def position_x(self) -> float:
-        """x-axis position of the graphic"""
-        return self.world_object.world.x
+        # set offset if it's not (0., 0., 0.)
+        if not all(self.world_object.world.position == self.offset):
+            self.offset = self.offset
 
-    @property
-    def position_y(self) -> float:
-        """y-axis position of the graphic"""
-        return self.world_object.world.y
+        # set rotation if it's not (0., 0., 0., 1.)
+        if not all(self.world_object.world.rotation == self.rotation):
+            self.rotation = self.rotation
+
+    def detach_feature(self, feature: str):
```

Comment:
Some methods (this one, but also e.g. `clear_event_handlers()` are missing a docstring.